.github/workflows/ci.yml

```diff
@@ -33,15 +33,48 @@ jobs:
           node-version: ${{ matrix.node }}
           architecture: ${{ matrix.arch }}
 
+      # FIXME: Install latest v8 once
+      # https://github.com/actions/setup-node/issues/411#issuecomment-1025543081
+      # is fixed.
+      #
+      # Even though we install a new version of node-gyp locally in
+      # the project, the CI always seems to use the npm bundled version of
+      # node-gyp. Even following the instructions from the docs, I could get it working
+      # on all other platforms except for Windows and npm@6, see
+      # - https://github.com/nodejs/node-gyp/blob/245cd5bbe4441d4f05e88f2fa20a86425419b6af/docs/Updating-npm-bundled-node-gyp.md
+      # - https://github.com/nodejs/node-gyp/blob/245cd5bbe4441d4f05e88f2fa20a86425419b6af/docs/Force-npm-to-use-global-node-gyp.md
+      - name: 'Install npm@^8 to get a more up to date bundled node-gyp'
+        run: npm install --global npm@8.3.1
+
+      # Fix `Error: Could not find any Visual Studio installation to use`
+      # See https://github.com/nodejs/node-gyp/tree/245cd5bbe4441d4f05e88f2fa20a86425419b6af/docs#issues-finding-the-installed-visual-studio
+      - name: 'Set msvs_version so node-gyp can find the Visual Studio install'
+        if: ${{ matrix.os == 'Windows' }}
+        run: npm config set msvs_version 2022
```

Comment:
This doesn't fix the `Error: Could not find any Visual Studio installation to use`, see https://github.com/MadLittleMods/node-usb-detection/runs/5400636691

But our `npm run node-gyp -- configure --verbose` command does seem to find it 🤔:

```
gyp verb find VS - looking for Visual Studio version 2022
gyp verb find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp verb find VS checking VS2022 (17.1.32210.238) found at:
gyp verb find VS "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
gyp verb find VS - found "Visual Studio C++ core features"
gyp verb find VS - found VC++ toolset: v1[43](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:43)
gyp verb find VS - found Windows SDK: 10.0.20348.0
gyp info find VS using VS2022 (17.1.32210.238) found at:
gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
```

Full output:
```
$ npm run node-gyp -- configure --verbose

> usb-detection@[4](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:4).13.0 node-gyp
> node-gyp "configure" "--verbose"

gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   'C:\\hostedtoolcache\\windows\\node\\14.19.0\\x[6](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:6)4\\node.exe',
gyp verb cli   'D:\\a\\node-usb-detection\\node-usb-detection\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'configure',
gyp verb cli   '--verbose'
gyp verb cli ]
gyp info using node-gyp@8.4.0
gyp info using node@14.19.0 | win32 | x64
gyp verb command configure []
gyp verb find Python Python is not set from command line or npm configuration
gyp verb find Python Python is not set from environment variable PYTHON
gyp verb find Python checking if "python3" can be used
gyp verb find Python - executing "python3" to get executable path
gyp verb find Python - executable path is "C:\hostedtoolcache\windows\Python\3.9.10\x64\python3.exe"
gyp verb find Python - executing "C:\hostedtoolcache\windows\Python\3.9.10\x64\python3.exe" to get version
gyp verb find Python - version is "3.9.10"
gyp info find Python using Python version 3.9.10 found at "C:\hostedtoolcache\windows\Python\3.9.10\x64\python3.exe"
gyp verb get node dir no --target version specified, falling back to host node version: 14.19.0
gyp verb command install [ '14.19.0' ]
gyp verb install input version string "14.19.0"
gyp verb install installing version: 14.19.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 14.19.0
gyp verb build dir attempting to create "build" dir: D:\a\node-usb-detection\node-usb-detection\build
gyp verb build dir "build" dir needed to be created? No
gyp verb find VS msvs_version was set from command line or npm config
gyp verb find VS - looking for Visual Studio version 2022
gyp verb find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp verb find VS checking VS2022 (1[7](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:7).1.32210.23[8](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:8)) found at:
gyp verb find VS "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
gyp verb find VS - found "Visual Studio C++ core features"
gyp verb find VS - found VC++ toolset: v143
gyp verb find VS - found Windows SDK: 10.0.20348.0
gyp info find VS using VS2022 (17.1.32210.238) found at:
gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
gyp info find VS run with --verbose for detailed information
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: D:\a\node-usb-detection\node-usb-detection\build\config.gypi
gyp verb config.gypi checking for gypi file: D:\a\node-usb-detection\node-usb-detection\config.gypi
gyp verb common.gypi checking for gypi file: D:\a\node-usb-detection\node-usb-detection\common.gypi
gyp verb gyp gyp format was not specified; forcing "msvs"
gyp info spawn C:\hostedtoolcache\windows\Python\3.[9](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:9).[10](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:10)\x64\python3.exe
gyp info spawn args [
gyp info spawn args   'D:\\a\\node-usb-detection\\node-usb-detection\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\a\\node-usb-detection\\node-usb-detection\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'D:\\a\\node-usb-detection\\node-usb-detection\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\runneradmin\\AppData\\Local\\node-gyp\\Cache\\[14](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:14).[19](https://github.com/MadLittleMods/node-usb-detection/runs/5400636691?check_suite_focus=true#step:11:19).0\\include\\node\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\runneradmin\\AppData\\Local\\node-gyp\\Cache\\14.19.0',
gyp info spawn args   '-Dnode_gyp_dir=D:\\a\\node-usb-detection\\node-usb-detection\\node_modules\\node-gyp',
gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\runneradmin\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\14.19.0\\\\<(target_arch)\\\\node.lib',
gyp info spawn args   '-Dmodule_root_dir=D:\\a\\node-usb-detection\\node-usb-detection',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'D:\\a\\node-usb-detection\\node-usb-detection\\build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok
```

Reply:
It looks like someone else is having trouble using `prebuild` in GitHub Actions:

> When you will update dependencies?
> It NOT works with visual studio 2022 on github actions.
>
> *-- Update dependencies: node-gyp stills using v6 it is currently in v8, https://github.com/prebuild/prebuild/issues/286#issuecomment-1049225288*

Reply:
Resorted to just reverting back to the `windows-2019` runners which work ✅